airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -201,7 +199,13 @@ private Throwable getRootConfigError(final Exception e) {
   }
 
   private boolean isConfigError(final Throwable e) {
-    return e instanceof ConfigErrorException || e instanceof ConnectionErrorException;
+    return e instanceof ConfigErrorException || e instanceof ConnectionErrorException || isRecoveryConnectionException(e);
+  }
+
+  private boolean isRecoveryConnectionException(Throwable e) {
+    return e instanceof SQLException && e.getMessage()
+            .toLowerCase(Locale.ROOT)
+            .contains("terminating connection due to conflict with recovery");
   }
 
   private String getDisplayMessage(final Throwable e) {
```

Comment:
We'd want to map this exception to a better error message (one that is not just the exception message). 
e.g. We're having issues syncing from a Postgres replica that is configured as a hot standby server. Please see https://docs.airbyte.com/integrations/sources/postgres/#sync-data-from-postgres-hot-standby-server for options and workarounds